### ISSUE TITLE
Suggested WebPack configuration breaks testing in (esnext) skeleton install...

### ISSUE THREAD

**Issue by svenvandescheur** [aurelia/framework#653]
After following [this](https://github.com/aurelia/framework/blob/master/doc/article/drafts/manual-webpack-configuration.md) tutorial on how to use the original WebPack configuration, the tests in the default (esnext) skeleton no longer work.

At first, webpack 2.1.0-beta.25 complains about the about this line:

`'app': [], // <-- this array will be filled by the aurelia-webpack-plugin`

Which is invalid in the new webpack version, uncommenting this makes the  tests move on, but only to show the real issue.

`Uncaught ReferenceError: webpackJsonp is not defined`

I first encountered this with a project I'm currently working on, which uses vanilla webpack configuration (no easy-webpack) and where I wanted to write tests for. This project isn't on the latest webpack so immediately showed the `Uncaught ReferenceError: webpackJsonp is not defined` error. I wanted to make sure it's not in my own project so tested it on the skeleton:

- Ran tests, all good
- Migrated following [this](https://github.com/aurelia/framework/blob/master/doc/article/drafts/manual-webpack-configuration.md) article.
- Ran tests, -> WebpackOptionsValidationError
- Uncommented "entry.app"
- Ran tests -> `Uncaught ReferenceError: webpackJsonp is not defined`

**I'm submitting a bug report**

* **Library Version:**
1.0.0

**Please tell us about your environment:**
* **Operating System:**
OSX 10.x

* **Node Version:**
6.2.2

* **NPM Version:**
3.9.5

* **JSPM OR Webpack AND Version**
webpack 2.1.0-beta.25

* **Browser:**
all

* **Language:**
ESNext

**Current behavior:**

```shell
npm test

> aurelia-skeleton-navigation-webpack@1.0.3 test /Users/svenvandescheur/Downloads/skeleton-navigation-1.0.3/skeleton-esnext-webpack
> cross-env NODE_ENV=test ./node_modules/karma/bin/karma start test/karma.conf.js


START:
10 11 2016 17:14:50.048:INFO [karma]: Karma v1.3.0 server started at http://localhost:9876/
10 11 2016 17:14:50.051:INFO [launcher]: Launching browser Chrome with unlimited concurrency
10 11 2016 17:14:50.080:INFO [launcher]: Starting browser Chrome
10 11 2016 17:14:51.436:INFO [Chrome 54.0.2840 (Mac OS X 10.11.6)]: Connected on socket /#_bzIS94DLPpkOqAGAAAA with id 64908103
Chrome 54.0.2840 (Mac OS X 10.11.6) ERROR
  Uncaught ReferenceError: webpackJsonp is not defined
  at spec-bundle.js:1
Chrome 54.0.2840 (Mac OS X 10.11.6) ERROR
  Uncaught ReferenceError: webpackJsonp is not defined
  at spec-bundle.js:1

Finished in 0.118 secs / 0 secs

SUMMARY:
✔ 0 tests completed
npm ERR! Test failed.  See above for more details.
```

**Expected/desired behavior:**

* **What is the expected behavior?**
Passing tests.

* **What is the motivation / use case for changing the behavior?**
Testing is awesome!

**Comment by svenvandescheur** [comment]
Some research points in the direction of CommonsChunkPlugin, I uncommented the section that uses it and everything "seems" to work again. Whats the purpose of this section?

**Comment by EisenbergEffect** [comment]
@niieani Can you chime in here?

**Comment by niieani** [comment]
@bigopon created the tutorial mentioned by @svenvandescheur.  

But to be fair, this seems like some sort of a Webpack issue, or a conflict in configuration, used plugins, etc. Hard to say without the full contents of the config file.

Can be closed as it's not Aurelia-related.

**Comment by bigopon** [comment]
@svenvandescheur You can drop both `app` in entry and the whole common chunks plugin if it's causing you troubles. Aurelia webpack plugin will still take care of your `src` folder nicely so no need to specify an entry point for your app modules.

I got common chunks sort the final entries in wrong order as well but I didn't have time to look into that so I once had this work around:

tl;dr: omitted additional `fetch.min.js`, `blurbird.min.js` and `fastclick.min.js` to dist folder, enabling full manual controls:
  * load CSS very early
  * polyfill on demand and only load app after polyfilled. (I had problem on IE when inlining bluebird / including in bundle didn't work because of wrong dependencies order, webpackJsonp)
  * reduce main bundle size

```html
<!--<meta name="bla" content="<%- JSON.stringify(htmlWebpackPlugin.files) %>"/>-->
    <link rel='stylesheet' href='<%- htmlWebpackPlugin.files.chunks.app.css[0] %>' />
    <script src='<%- htmlWebpackPlugin.files.chunks.polyfills.entry %>'></script>
    <script type='text/javascript' id='entry-script'>
      (function () {
        function appendScript() {
          var args = arguments
          var baseurl = arguments[0]
          var scriptTag = document.createElement('script')
          scriptTag.src = baseurl
          if (arguments[1] && typeof arguments[1] === 'string') {
            scriptTag.onload = function() {
              scriptTag.onload = null
              var tobeLoaded = Array.prototype.slice.call(args, 1)
              appendScript.apply(null, tobeLoaded)
            }
          }
          document.head.appendChild(scriptTag)
        }
        var script, today = +new Date();
        if (!window.fetch) {
          script = document.createElement('script')
          script.src = 'fetch.min.js?' + today
          document.head.appendChild(script)
        }
        var isMobile = /iPhone|iPad|iPod|Android/i.test(navigator.userAgent);
        if (isMobile) {
          script = document.createElement('script')
          script.src = 'fastclick.min.js?' + today
          script.onload = function() {
            script.onload = null
            if (document.readyState === 'complete')
              FastClick.attach(document.body)
            else
              document.addEventListener('DOMContentLoaded', function() {
                FastClick.attach(document.body);
              }, false);
          }
          document.head.appendChild(script)
        }
        function appendAllScripts() {
          appendScript('<%- htmlWebpackPlugin.files.chunks.aurelia.entry %>', '<%- htmlWebpackPlugin.files.chunks.app.entry %>')
        }
        if (!window.Promise) {
          script = document.createElement('script')
          script.type = 'text/javascript'
          script.src = 'bluebird.min.js?' + today
          script.onload = function() {
            script.onload = null
            appendAllScripts()
          }
          document.head.appendChild(script)
        } else {
          appendAllScripts()
        }
      })()
    </script>
```
